CRM/Activity/BAO/Activity.php

```diff
@@ -840,10 +808,10 @@ public static function getActivities($params, $getCount = FALSE) {
         else {
           $activities[$id][$expectedName] = CRM_Utils_Array::value($apiKey, $activity);
           if ($apiKey == 'activity_type_id') {
-            $activities[$id]['activity_type'] = CRM_Utils_Array::value($activities[$id][$expectedName], $activityTypes);
+            $activities[$id]['activity_type'] = CRM_Core_PseudoConstant::getName(__CLASS__, 'activity_type_id', $activities[$id][$expectedName]);
           }
           elseif ($apiKey == 'campaign_id') {
-            $activities[$id]['campaign'] = CRM_Utils_Array::value($activities[$id][$expectedName], $allCampaigns);
```

Comment:
I actually feel a bit nervous about this line - as an unrelated change I feel like I have to work quite hard to validate it or let it skate through (& the latter is not good)

Reply:
Ok, changed back to original.